core-rust/state-manager/src/protocol/protocol_config.rs

```diff
@@ -0,0 +1,178 @@
+use radix_engine::prelude::{dec, ScryptoCategorize, ScryptoDecode, ScryptoEncode};
+use radix_engine_common::math::Decimal;
+use radix_engine_common::prelude::{hash, scrypto_encode};
+
+use radix_engine_common::types::Epoch;
+
+use crate::ProtocolUpdaterFactory;
+use utils::btreeset;
+
+// This file contains types for node's local static protocol configuration
+
+pub const GENESIS_PROTOCOL_VERSION: &str = "babylon-genesis";
+pub const ANEMONE_PROTOCOL_VERSION: &str = "anemone";
+
+const MAX_PROTOCOL_VERSION_NAME_LEN: usize = 16;
+
+/// Returns a protocol version name left padded to canonical length (16 bytes)
+pub fn padded_protocol_version_name(unpadded_protocol_version_name: &str) -> String {
+    let mut res = "".to_owned();
+    for _ in 0..16 - unpadded_protocol_version_name.len() {
+        res.push('0');
+    }
+    res.push_str(unpadded_protocol_version_name);
+    res
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub struct ProtocolUpdate {
+    pub next_protocol_version: String,
+    pub enactment_condition: ProtocolUpdateEnactmentCondition,
+}
+
+impl ProtocolUpdate {
+    pub fn readiness_signal_name(&self) -> String {
+        // Readiness signal name is 32 bytes long and consists of:
+        // - 16 hexadecimal chars of leading bytes of `hash(enactment_condition + next_protocol_version)`
+        // - next_protocol_version: 16 bytes,
+        //      left padded with ASCII 0's if protocol version name is shorter than 16 characters
+        let mut bytes_to_hash = scrypto_encode(&self.enactment_condition).unwrap();
+        bytes_to_hash.extend_from_slice(self.next_protocol_version.as_bytes());
+        let protocol_update_hash = hash(&bytes_to_hash);
+        let mut res = hex::encode(protocol_update_hash)[0..16].to_string();
+        res.push_str(&padded_protocol_version_name(&self.next_protocol_version));
+        res
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub struct ProtocolConfig {
+    pub genesis_protocol_version: String,
+    pub protocol_updates: Vec<ProtocolUpdate>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub enum ProtocolUpdateEnactmentCondition {
```

Comment:
Perhaps just `ProtocolUpdateCondition`?

Reply:
I don't have a strong opinion. What do others think? @jakrawcz-rdx @dhedey ?

Reply:
If you use "enactment, enact, enacted" widely throughout this feature, then `Enactment` part here is very relevant. And long names are actually our convention ;p
However, I've seen some discussions about other renames (like, "unenacted -> pending" [here](https://github.com/radixdlt/babylon-node/pull/790#discussion_r1452829020)). If you go this way, then I am for removing `Enactment` from here too.

Reply:
Slight preference for `ProtocolUpdateEnactmentCondition` but I don't mind really.

I think `protocol_update.enactment_condition` reads slightly better than `protocol_update.condition` but again, not a massive deal either way.